streams/src/main/java/org/apache/kafka/streams/kstream/internals/WindowedDeserializer.java

```diff
@@ -64,13 +72,15 @@ public void configure(Map<String, ?> configs, boolean isKey) {
     public Windowed<T> deserialize(String topic, byte[] data) {
 
         byte[] bytes = new byte[data.length - TIMESTAMP_SIZE];
-
+        
+        System.out.println("deserialize method cDs.alled");
         System.arraycopy(data, 0, bytes, 0, bytes.length);
-
+        
         long start = ByteBuffer.wrap(data).getLong(data.length - TIMESTAMP_SIZE);
-
-        // always read as unlimited window
-        return new Windowed<T>(inner.deserialize(topic, bytes), new UnlimitedWindow(start));
+        
+        return windowSize != Long.MAX_VALUE ? new Windowed<T>(inner.deserialize(topic, bytes), 
```

Comment:
IMHO a little hard to follow and some logic is duplicated,  what about something like
```
Window timeWindow = windowSize != Long.MAX_VALUE ? WindowStoreUtils.timeWindowForSize(start, windowSize) : new UnlimitedWindow(start);

return new Windowed<T>(inner.deserialize(topic, bytes), timeWindow);
```

Reply:
It should be noted that the Window type will also have to be imported, since it was not used
anywhere else in the code. I would use TimeWindow type instead for declaration.